src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.ValidationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String FIRST_VALID_LOGIN = "Username";
+    private static final String SECOND_VALID_LOGIN = "Username2";
+    private static final String VALID_PASSWORD = "Password";
+    private static final String INVALID_LOGIN = "12345";
+    private static final String INVALID_PASSWORD = "12345";
+    private static final int VALID_AGE = 18;
+    private static long id;
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User validUser = new User();
+    private User existUser = validUser;
+
+    @BeforeEach
+    void beforeEach() {
+        validUser.setAge(VALID_AGE);
+        validUser.setLogin(FIRST_VALID_LOGIN);
+        validUser.setPassword(VALID_PASSWORD);
+        validUser.setId(id++);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.PEOPLE.clear();
+    }
+
```

Comment:
general recommendation regarding structuring your tests - try to group them:
1. successful tests
2. login tests (empty, null, longer, shorter, used etc.)
3. password tests
4. age tests

this way it will be easier for reader to find a corresponding test scenario